doc/topics/writing-recipes.rst

```diff
@@ -337,6 +342,267 @@ entry points, using the type ``zc.buildout.uninstall`` as can be seen
 in the `zc.recipe.rhrc setup script
 <https://github.com/zopefoundation/zc.recipe.rhrc/blob/master/setup.py#L23>`_.
 
+Testing recipes
+================
+
+The recipe API is fairly simple and standard unit-testing approaches
+can be used.  We'll use a helper class,
+``zc.buildout.testing.Buildout`` [#helper-class-refined-in-2.9]_ to
+provide a minimal buildout environment.
+
+Let's write a test for our configuration recipe.  We need to verify that:
+
+- The recipe generates a ``path`` option.
+
+- The recipe generates a file in the correct place.
+
+- The recipe returns the path it created from ``install``.
+
+.. _recipe-example:
+
+We create a ``testdemoconfigrecipe.py`` file containing our tests::
+
+  import os
+  import shutil
+  import tempfile
+  import unittest
+  import zc.buildout.testing
+
+  class RecipeTests(unittest.TestCase):
+
+      def setUp(self):
+          self.here = os.getcwd()
+          self.tmp = tempfile.mkdtemp()
```

Comment:
I always recommend passing a prefix to `mkdtemp()`, to make it easier to track down the source in case some test forgets to clean up:
```
self.tmp = tempfile.mkdtemp(prefix='testdemoconfigrecipe-')
```